byzcoin/service.go

```diff
@@ -2582,10 +2673,9 @@ func (s *Service) fixInconsistencyIfAny(genesisID skipchain.SkipBlockID, st *sta
 		return err
 	}
 
-	var header DataHeader
-	err = protobuf.Decode(currSB.Data, &header)
+	header, err := decodeBlockHeader(currSB)
 	if err != nil {
-		return errors.New("couldn't unmarshal header: " + err.Error())
+		return err
```

Comment:
Please be careful when removing these "stacked errors", as it can be difficult to track down the error when you only have the error from `decodeBlockHeader`. As the `decodeBlockHeader` is called multiple times, if you find the error in the logs, you don't know where it failed. This is why these `return errors.New("more error details" + err.Error())` are useful.